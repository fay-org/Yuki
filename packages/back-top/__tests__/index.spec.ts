import BackTop from '..'
import { mount, flushPromises } from '@vue/test-utils'
import { later } from '../../../tests'

const CustomWrapper = {
  components:{
    [BackTop.name]:BackTop,
  },
  data(){
    return {
      right: 50,
      bottom: 50,
      visibilityHeight: 10,
    }
  },
  template:`<div class="container" ref="scrollTarget" style="height: 100px; overflow: auto">
    <div style="height: 10000px; width: 100%;">
      <fe-back-top ref="target" :right="right" :bottom="bottom" :visibilityHeight="visibilityHeight" >
        Custom Context
      </fe-back-top>
    </div>
  </div>`,
}

const Wrapper = {
  components:{
    [BackTop.name]:BackTop,
  },
  data(){
    return {
      duration: 1,
      right: 50,
      bottom: 50,
    }
  },
  template:`<div class="container" ref="scrollTarget" style="height: 100px; overflow: auto">
    <div style="height: 10000px; width: 100%;">
      <fe-back-top ref="target" :target="() => $refs.scrollTarget" :duration="duration" :right="right" :bottom="bottom" :visibilityHeight="visibilityHeight" >
        Custom Context
      </fe-back-top>
    </div>
  </div>`,
}

describe('BackTop', () => {
  it('should be render as element', () => {
    const wrapper = mount(BackTop, { 
      props: {
        visibilityHeight: 10,
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })
  it('should support props right and bottom', () => {
    const wrapper = mount(BackTop, { 
      props: {
        right: 50,
        bottom: 50,
        visibilityHeight: 10,
      },
    })
    const el = wrapper.find('.fect-back-top')
    expect(el.attributes('style')).toBe(
      'right: 50px; bottom: 50px;',
    )
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('should be support custom slots',async () => {
    const wrapper = mount(CustomWrapper)
    const el = wrapper.find('.fect-back-top')
    expect(el.text()).toBe('Custom Context')
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('should be support scroll visible',async () => {
    const wrapper = mount(Wrapper)
    const el = wrapper.find('.fect-back-top')
    expect(el.text()).toBe('')

    const refs = wrapper.vm.$refs as any
    const {
      target: { handleScroll },
    } = refs
    refs.scrollTarget.scrollTop = 2000
    await handleScroll({ target: refs.scrollTarget })
    await flushPromises()
    await later()
    expect(el.text()).toBe('Custom Context')

    await el.trigger('click')
    await flushPromises()
    await later()
    await handleScroll({ target: refs.scrollTarget })
    await flushPromises()
    await later()
    expect(el.text()).toBe('')
    expect(wrapper.html()).toMatchSnapshot()
  })
})
