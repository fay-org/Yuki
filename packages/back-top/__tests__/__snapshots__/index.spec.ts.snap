// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackTop should be render as element 1`] = `
<div class="fect-back-top" style="right: 40px; bottom: 40px;">
  <div class="fect-back-top__content"><span class="fect-back-top__icon"><svg viewBox="64 64 896 896" width="1.5rem" height="1.5rem" fill="currentColor" aria-hidden="true"><path d="M859.9 168H164.1c-4.5 0-8.1 3.6-8.1 8v60c0 4.4 3.6 8 8.1 8h695.8c4.5 0 8.1-3.6 8.1-8v-60c0-4.4-3.6-8-8.1-8zM518.3 355a8 8 0 00-12.6 0l-112 141.7a7.98 7.98 0 006.3 12.9h73.9V848c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V509.7H624c6.7 0 10.4-7.7 6.3-12.9L518.3 355z"></path></svg></span></div>
</div>
`;

exports[`BackTop should be support custom slots 1`] = `
<div class="container" style="height: 100px; overflow: auto;">
  <div style="height: 10000px; width: 100%;">
    <div class="fect-back-top" style="right: 50px; bottom: 50px;"> Custom Context </div>
  </div>
</div>
`;

exports[`BackTop should be support scroll visible 1`] = `
<div class="container" style="height: 100px; overflow: auto;">
  <div style="height: 10000px; width: 100%;">
    <div class="fect-back-top" style="right: 50px; bottom: 50px;">
      <!---->
    </div>
  </div>
</div>
`;

exports[`BackTop should support props right and bottom 1`] = `
<div class="fect-back-top" style="right: 50px; bottom: 50px;">
  <div class="fect-back-top__content"><span class="fect-back-top__icon"><svg viewBox="64 64 896 896" width="1.5rem" height="1.5rem" fill="currentColor" aria-hidden="true"><path d="M859.9 168H164.1c-4.5 0-8.1 3.6-8.1 8v60c0 4.4 3.6 8 8.1 8h695.8c4.5 0 8.1-3.6 8.1-8v-60c0-4.4-3.6-8-8.1-8zM518.3 355a8 8 0 00-12.6 0l-112 141.7a7.98 7.98 0 006.3 12.9h73.9V848c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V509.7H624c6.7 0 10.4-7.7 6.3-12.9L518.3 355z"></path></svg></span></div>
</div>
`;
